### ISSUE TITLE
Let-binding during creation, in combination with `runresult` on strings

### ISSUE THREAD

**Issue by arthurhjorth** [NetLogo/NetLogo#1728]
I think I found a weird bug. But let-binding is definitely on the periphery of things that I understand, so maybe this is intended. I dont think so though.

running  (i.e. calling `test` from the command center) this code

```
to test
  ca
  crt 1 [
    let avar random 100
      if avar > 50 [
        let rando random 100
      ]
      if avar  < 50 [
        let rando random 100
      ]
    ]
  show map [ r -> runresult r] reporters
end

to-report reporters
  report (list "count turtles" "count links" )
end

```

results in a runtime error in which `runresult` is highlighted and you get a pop-up stating that `There is already a local variable called RANDO here`. But the scope of each of the `rando` let-vars  should just be each of the if-brackets, right? So that seems odd to me.

If you create the turtles first, and then `ask` them to run the command block  (rather than passing it into `create-`) , you do not get this error. 

If you only have one `if` bracket (i.e. if you only create `rando` once), then you do not get it.

If you define the let-var before the `if` brackets, and then use `set` on it, then you do not get it.

Is this a bug? any idea what the problem is?

**Comment by SethTisue** [comment]
I'm not able to reproduce the problem in either 6.0.4 or 6.1.0-RC1

**Comment by arthurhjorth** [comment]
Really? That is so odd... If I just copy paste this code into the code tab on a new model in NetLogo 6.0.4 or 6.1.0-RC1 and then call **test** from the command line, it happens. This is on both Windows 10 and on Mac OS Mojave.

I will try on my computer at home when I get home from the office.

**Comment by SethTisue** [comment]
ohhhhhh.... I thought you meant a compile time error. I didn't try running the code :-)

never mind

**Comment by SethTisue** [comment]
yes, absolutely a bug.  (`run` and `runresult` on strings can be pretty janky, but something as simple as `"count turtles"` really ought to work no matter what)

**Comment by arthurhjorth** [comment]
Oh sorry, yes! It's a runtime error. Edited the issue to reflect that :)

**Comment by charlesres** [comment]
Seems like a possible duplicate of https://github.com/NetLogo/NetLogo/issues/600. I was able to reproduce it with NetLogo 6.1.0-RC1. There's a pull request that fixes this issue, which works with the code above.

**Comment by LaCuneta** [comment]
I confirmed this is fixed by the PR for #600 in 6.1.1-RC1.
